static/js/stream_settings_data.js

```diff
@@ -60,11 +60,6 @@ export function update_calculated_fields(sub) {
     sub.preview_url = hash_util.by_stream_uri(sub.stream_id);
 
     sub.is_old_stream = sub.stream_weekly_traffic !== null;
-
-    // Apply the defaults for our notification settings for rendering.
-    for (const setting of settings_config.stream_specific_notification_settings) {
-        sub[setting + "_display"] = stream_data.receives_notifications(sub.stream_id, setting);
-    }
```

Comment:
I believe its intent is to handle the specification for how our stream-level notification settings work: That if one hasn't set a specific value for a stream, you inherit the setting from the user's global notification settings.  While `stream_data.receives_notifications` handles that calculation for actually controlling the notifications, we need to use this calculation to display whether the setting is checked or not in the notification settings table.

E.g. `UserProfile.enable_stream_desktop_notifications = true`, `Subscription.dekstop_notifications = None` => we need to display the box as checked.  

I think what happened is that `get_unmatched_streams_for_notification_settings` now implements this functionality, via the `stream_data.receives_notifications` call there, and no longer reads these fields; I think that's the correct design, since it also needs to inspect the computed values for other reasons.  

So I think you should just update the commit message for this to mention it was made obsolete by c759083cb891f24a3b90b96d782c0ee7b9ab7f7d. 

@ryanreh99 FYI to help confirm this story.

Reply:
As discussed over chat, the code isn't dead, but we can have `stream_edit.js` just call the same function that we used for the crufty `sub.foo_display` calculation.